canonical/charmhub.io
# Create discourse search endpoint

## Done
- create endpoint for searching discourse API
- add flask-caching to requirements.txt file
## How to QA
- Go to https://charmhub-io-1643.demos.haus/docs?q={search-term} and ensure that it returns a response containing a list of posts and a list of topics. Each list should have four items each
- Go to https://charmhub-io-1643.demos.haus/docs/all-posts?q={search-term}&page=page and ensure that it returns a response containing a list of posts and total-pages
- Go to https://charmhub-io-1643.demos.haus/docs/all-topics?q={search-term}?page=page and ensure that it returns a response containing a list of topics and total-pages
- Example of search-terms include: `juju`, `charm`, `database`, `k8s` and so on
- Searching for an invalid search term like `a` should return a response containing an error key and status code of not found
## Issue / Card https://warthogs.atlassian.net/browse/WD-5268
Fixes #

## Screenshots
